### ISSUE TITLE
`networkctl reload` did not apply new prefix length for address

### ISSUE THREAD

**Issue by anitazha** [systemd/systemd#22515]
**systemd version the issue has been seen with**
 > v249.4

**Used distribution**
 > CentOS 8 stream

**Linux kernel version used** (`uname -a`)
> `5.6.13` with patches

I have a config that looks like this:
```
$ cat /etc/systemd/network/1-fb_networkd-eth0.network 
[Match]
MACAddress = 48:57:DD:05:C8:DC
Name = eth0

[Network]
Address = <primary address>/64
IPv4LLRoute = true
IPv6AcceptRA = true

[IPv6AcceptRA]
UseMTU = false
UseAutonomousPrefix = false
UseOnLinkPrefix = false
DHCPv6Client = false
```

I apply the config with `networkctl reload && networkctl reconfigure eth0` and `ip addr` shows the expected address with prefix length 64.

But if I change the line in the config from `Address = <primary address>/64` to `Address = <primary address>/128` (i.e. only the prefix length has changed), and do `networkctl reload` and `networkctl reconfigure eth0`, the address and prefix length as shown by `ip addr` is not updated. The only way I successfully got the new prefix length to apply is if I do `networkctl down eth0 && networkctl up eth0`. Is this expected?

**Comment by yuwata** [comment]
Yeah, I know the issue...

Previously, once I changed the hash function for addresses to takes into account prefix length of address, but it causes an issue (though I forgot the detail...). So, now the function hashes only address. I will re-investigate the previous issue and this later.

Note, the issue should be occurs only on IPv6, as the prefix length of IPv4 addresses is already managed by the hash function.

**Comment by yuwata** [comment]
Note, the previous change is 1d30fc5cb64ecba2f03fe42aa0d8c65c3decad82.

**Comment by yuwata** [comment]
Fix is waiting in #22607.

**Comment by anitazha** [comment]
Thanks @yuwata ! Your PR worked like a charm
